packages/test/test-version-utils/src/test/versionUtils.spec.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { getRequestedRange } from "../versionUtils";
+
+describe("Get the major version number above or below the baseVersion", () => {
+    // assert for major bumps
+    assert.strictEqual(getRequestedRange("1.0.0", -1), "^0.59.0-0");
+    assert.strictEqual(getRequestedRange("1.0.0", -2), "^0.58.0-0");
+    assert.strictEqual(getRequestedRange("1.0.0", 1), "^2.0.0-0");
+    assert.strictEqual(getRequestedRange("2.0.0", -1), "^1.0.0-0");
+
+    // assert for internal release
+    assert.strictEqual(getRequestedRange("2.0.0-internal.1.0.0", -1), "^1.0.0-0");
```

Comment:
maybe add some other test cases?

2.0.0-internal.1.2.3 => 1.0.0-0
2.0.0-internal.2.0.0 => 2.0.0-internal.1.0.0
2.0.0-internal.2.1.0 => 2.0.0-internal.1.0.0
2.0.0-internal.3.0.0 => 2.0.0-internal.2.0.0

Reply:
added these in